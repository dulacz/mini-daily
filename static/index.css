/* Main Content Grid - Responsive 3 columns */
.main-content {
    display: grid;
    grid-template-columns: 1fr;
    gap: 1.5rem;
    margin-bottom: 2rem;
}

/* Desktop: 3 columns */
@media (min-width: 1024px) {
    .main-content {
        grid-template-columns: repeat(3, 1fr);
    }
}

/* Tablet: 2 columns */
@media (min-width: 768px) and (max-width: 1023px) {
    .main-content {
        grid-template-columns: repeat(2, 1fr);
    }
}

/* Activity Grid Layout (3x3) */
.activity-grid {
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    gap: 0.75rem;
    margin-top: 1rem;
}

.activity-btn-grid {
    background: var(--surface-color);
    border: 2px solid var(--border-color);
    border-radius: var(--radius-md);
    padding: 1rem 0.5rem;
    cursor: pointer;
    transition: all 0.3s ease;
    text-align: center;
    font-size: 0.875rem;
    font-weight: 500;
    color: var(--text-primary);
    min-height: 80px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.activity-btn-grid:hover {
    border-color: var(--primary-color);
    background: rgba(99, 102, 241, 0.05);
    transform: scale(1.02);
}

.activity-btn-grid.active {
    border-color: var(--success-color);
    background: var(--success-color);
    color: white;
    font-weight: 600;
}

.activity-btn-grid.active:hover {
    background: #059669;
    border-color: #059669;
}

.activity-btn-grid .activity-label {
    font-weight: 600;
    font-size: 0.875rem;
}

.activity-btn-grid .activity-last-completed {
    font-size: 0.7rem;
    opacity: 0.6;
    margin-top: 0.375rem;
    font-weight: 400;
    color: rgba(255, 255, 255, 0.7);
}

/* Loading and Error States */
.loading-state,
.error-state {
    text-align: center;
    padding: 2rem;
    color: var(--text-secondary);
}

.loading-spinner {
    width: 40px;
    height: 40px;
    border: 4px solid var(--border-color);
    border-top-color: var(--primary-color);
    border-radius: 50%;
    animation: spin 1s linear infinite;
    margin: 0 auto 1rem;
}

@keyframes spin {
    to {
        transform: rotate(360deg);
    }
}

.retry-btn {
    background: var(--primary-color);
    color: white;
    border: none;
    padding: 0.75rem 1.5rem;
    border-radius: var(--radius-md);
    font-weight: 500;
    cursor: pointer;
    transition: background 0.2s ease;
}

.retry-btn:hover {
    background: var(--secondary-color);
}

/* Historical Calendar Styles */
.history-section {
    margin-top: 2rem;
    padding: 1.5rem;
    background: var(--surface-color);
    border-radius: var(--radius-lg);
    border: 1px solid var(--border-color);
}

.history-header {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 1.5rem;
}

.month-navigation {
    display: flex;
    align-items: center;
    gap: 1rem;
}

.month-nav-btn {
    background: var(--surface-color);
    border: 2px solid var(--border-color);
    border-radius: var(--radius-md);
    padding: 0.5rem 1rem;
    cursor: pointer;
    font-size: 1.25rem;
    color: var(--text-primary);
    transition: all 0.2s ease;
}

.month-nav-btn:hover {
    border-color: var(--primary-color);
    background: rgba(99, 102, 241, 0.05);
}

.current-month {
    font-size: 1.125rem;
    font-weight: 600;
    color: var(--text-primary);
    min-width: 150px;
    text-align: center;
}

.calendar-header {
    display: grid;
    grid-template-columns: repeat(7, 1fr);
    gap: 0.35rem;
    margin-bottom: 0.5rem;
}

.day-header {
    text-align: center;
    font-weight: 600;
    font-size: 0.875rem;
    color: var(--text-secondary);
    padding: 0.5rem;
}

.calendar-grid {
    display: grid;
    grid-template-columns: repeat(7, 1fr);
    gap: 0.35rem;
}

.task-stats {
    display: flex;
    align-items: center;
    gap: 0.25rem;
}

/* Responsive adjustments for activity grid */
@media (max-width: 767px) {
    .activity-grid {
        gap: 0.5rem;
    }

    .activity-btn-grid {
        padding: 0.75rem 0.25rem;
        font-size: 0.75rem;
        min-height: 50px;
    }

    /* Mobile calendar adjustments for better fit */
    .calendar-grid {
        gap: 0.15rem !important;
    }

    .day-cell {
        padding: 0.15rem !important;
        min-height: auto !important;
        font-size: 0.7rem !important;
    }

    .day-number {
        font-size: 0.65rem !important;
        line-height: 1 !important;
    }

    .day-stars {
        font-size: 0.55rem !important;
        margin-top: 0.1rem !important;
        line-height: 1 !important;
    }

    .calendar-header {
        gap: 0.15rem !important;
        margin-bottom: 0.2rem !important;
    }

    .day-header {
        padding: 0.2rem !important;
        font-size: 0.65rem !important;
    }

    .day-cell:hover:not(.outside-month) {
        transform: scale(1.02) !important;
    }
}

/* Extra small screens (phones) */
@media (max-width: 480px) {
    .calendar-grid {
        gap: 0.1rem !important;
    }

    .day-cell {
        padding: 0.1rem !important;
        border-width: 1px !important;
    }

    .day-number {
        font-size: 0.6rem !important;
    }

    .day-stars {
        font-size: 0.5rem !important;
        margin-top: 0.05rem !important;
    }

    .day-header {
        padding: 0.15rem !important;
        font-size: 0.6rem !important;
    }

    .calendar-header {
        gap: 0.1rem !important;
        margin-bottom: 0.15rem !important;
    }
}
