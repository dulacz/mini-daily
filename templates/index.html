<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta name="description"
        content="Happy coding and have fun, generated by pmvibecoding template - https://github.com/chenxizhang/pmvibecoding, empower every product manager on the planet to achieve more by building their product prototype with AI.">
    <title>Daily Check-in</title>
    <link rel="stylesheet" href="{{ url_for('static', path='style.css') }}">
    <link rel="icon" type="image/svg+xml" href="{{ url_for('static', path='favicon.svg') }}" />
    <link rel="preconnect" href="https://fonts.googleapis.com">
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
    <link href="https://fonts.googleapis.com/css2?family=Inter:wght@400;500;600;700&display=swap" rel="stylesheet">
</head>

<body>
    <!-- Animated Background Particles -->
    <div class="particles" id="particles"></div>

    <!-- Theme Toggle -->
    <div class="theme-toggle" id="themeToggle" title="Toggle theme">
        üåô
    </div>

    <div class="app-container">
        <!-- User Tabs -->
        <section class="user-tabs">
            <div class="tab-container" id="userTabs">
                <!-- User tabs will be generated dynamically by JavaScript -->
            </div>
        </section>

        <!-- Header Section -->
        <header class="header">
            <div class="header-content">
                <h1 class="app-title" id="appTitle">Daily Check-in</h1>
                <div class="streak-counter">
                    <span class="streak-icon">üî•</span>
                    <span class="streak-text">
                        <span class="streak-number" id="streakCount">0</span>
                        <span class="streak-label">day streak</span>
                    </span>
                </div>
                <div class="star-counter">
                    <span class="star-icon">‚≠ê</span>
                    <span class="star-text">
                        <span class="star-number" id="starCount">0</span>
                        <span class="star-label">total stars</span>
                    </span>
                </div>
            </div>
            <div class="date-display" id="dateDisplay">{{ today }}</div>
        </header>

        <!-- Progress Overview -->
        <section class="progress-overview">
            <div class="progress-circle">
                <svg class="progress-ring" width="120" height="120">
                    <circle class="progress-ring-bg" cx="60" cy="60" r="50"></circle>
                    <circle class="progress-ring-fill" cx="60" cy="60" r="50" id="progressCircle"></circle>
                </svg>
                <div class="progress-text">
                    <span class="progress-percentage" id="progressPercentage">0%</span>
                    <span class="progress-label">Complete</span>
                </div>
            </div>
            <div class="daily-stats">
                <div class="stat-summary">
                    <span class="stat-number" id="totalStars">0</span><span class="stat-icon">‚≠ê</span><span
                        class="stat-separator"></span><span class="stat-number" id="completedTasks">0</span> <span
                        class="stat-text">Tasks Done</span>
                </div>
            </div>
        </section>

        <!-- Main Tasks Section -->
        <main class="tasks-container" id="tasksContainer">
            <!-- Task cards will be generated dynamically by JavaScript -->
        </main>

        <!-- Achievement Section -->
        <section class="achievements" id="achievementsSection" style="display: none;">
            <div class="achievement-popup" id="achievementPopup">
                <div class="achievement-content">
                    <div class="achievement-icon">üéâ</div>
                    <h3 class="achievement-title" id="achievementTitle"></h3>
                    <p class="achievement-message" id="achievementMessage"></p>
                </div>
            </div>
        </section>

        <!-- Monthly Summary (initially hidden) -->
        <section class="monthly-summary">
            <button class="summary-toggle" id="monthlySummaryToggle">
                <span>View Monthly Progress</span>
                <span class="toggle-icon">üìÖ</span>
            </button>
            <div class="summary-content" id="monthlySummaryContent" style="display: none;">
                <h3 id="monthlyTitle">This Month's Progress</h3>
                <div class="monthly-grid" id="monthlyGrid"></div>
            </div>
        </section>
    </div>

    <script src="{{ url_for('static', path='script.js') }}"></script>
</body>

</html>