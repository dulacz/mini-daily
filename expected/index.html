<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta name="description"
        content="Happy coding and have fun, generated by pmvibecoding template - https://github.com/chenxizhang/pmvibecoding, empower every product manager on the planet to achieve more by building their product prototype with AI.">
    <title>Daily Wellness Tracker</title>
    <link rel="stylesheet" href="styles.css">
    <link rel="preconnect" href="https://fonts.googleapis.com">
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
    <link href="https://fonts.googleapis.com/css2?family=Inter:wght@400;500;600;700&display=swap" rel="stylesheet">
</head>

<body>
    <div class="app-container">
        <!-- Header Section -->
        <header class="header">
            <div class="header-content">
                <h1 class="app-title">Daily Wellness</h1>
                <div class="streak-counter">
                    <span class="streak-icon">üî•</span>
                    <span class="streak-text">
                        <span class="streak-number" id="streakCount">0</span>
                        <span class="streak-label">day streak</span>
                    </span>
                </div>
            </div>
            <div class="date-display" id="dateDisplay"></div>
        </header>

        <!-- Progress Overview -->
        <section class="progress-overview">
            <div class="progress-circle">
                <svg class="progress-ring" width="120" height="120">
                    <circle class="progress-ring-bg" cx="60" cy="60" r="50"></circle>
                    <circle class="progress-ring-fill" cx="60" cy="60" r="50" id="progressCircle"></circle>
                </svg>
                <div class="progress-text">
                    <span class="progress-percentage" id="progressPercentage">0%</span>
                    <span class="progress-label">Complete</span>
                </div>
            </div>
            <div class="daily-stats">
                <div class="stat-item">
                    <span class="stat-value" id="totalStars">0</span>
                    <span class="stat-label">Stars Today</span>
                </div>
                <div class="stat-item">
                    <span class="stat-value" id="completedTasks">0</span>
                    <span class="stat-label">Tasks Done</span>
                </div>
            </div>
        </section>

        <!-- Main Tasks Section -->
        <main class="tasks-container">
            <div class="task-card" data-task="reading">
                <div class="task-header">
                    <div class="task-icon">üìö</div>
                    <h3 class="task-title">Reading</h3>
                    <div class="task-badge" id="readingBadge">Not Started</div>
                </div>
                <div class="task-description">
                    <span class="task-desc-text">Feed your mind with knowledge</span>
                </div>
                <div class="star-rating" data-task="reading">
                    <button class="star-btn" data-level="1" title="1 page read">
                        <span class="star-icon">‚≠ê</span>
                        <span class="star-label">1 page</span>
                    </button>
                    <button class="star-btn" data-level="2" title="5 minutes reading">
                        <span class="star-icon">‚≠ê</span>
                        <span class="star-label">5 min</span>
                    </button>
                    <button class="star-btn" data-level="3" title="15 minutes reading">
                        <span class="star-icon">‚≠ê</span>
                        <span class="star-label">15 min</span>
                    </button>
                </div>
                <div class="task-streak">
                    <span class="task-streak-text">Streak: <span class="task-streak-count" id="readingStreak">0</span>
                        days</span>
                </div>
            </div>

            <div class="task-card" data-task="exercise">
                <div class="task-header">
                    <div class="task-icon">üí™</div>
                    <h3 class="task-title">Exercise</h3>
                    <div class="task-badge" id="exerciseBadge">Not Started</div>
                </div>
                <div class="task-description">
                    <span class="task-desc-text">Strengthen your body and mind</span>
                </div>
                <div class="star-rating" data-task="exercise">
                    <button class="star-btn" data-level="1" title="10 minutes movement">
                        <span class="star-icon">‚≠ê</span>
                        <span class="star-label">10 min</span>
                    </button>
                    <button class="star-btn" data-level="2" title="20 minutes workout">
                        <span class="star-icon">‚≠ê</span>
                        <span class="star-label">20 min</span>
                    </button>
                    <button class="star-btn" data-level="3" title="45 minutes exercise">
                        <span class="star-icon">‚≠ê</span>
                        <span class="star-label">45 min</span>
                    </button>
                </div>
                <div class="task-streak">
                    <span class="task-streak-text">Streak: <span class="task-streak-count" id="exerciseStreak">0</span>
                        days</span>
                </div>
            </div>

            <div class="task-card" data-task="caring">
                <div class="task-header">
                    <div class="task-icon">‚ù§Ô∏è</div>
                    <h3 class="task-title">Self-Care</h3>
                    <div class="task-badge" id="caringBadge">Not Started</div>
                </div>
                <div class="task-description">
                    <span class="task-desc-text">Nurture your wellbeing</span>
                </div>
                <div class="star-rating" data-task="caring">
                    <button class="star-btn" data-level="1" title="5 minutes meditation">
                        <span class="star-icon">‚≠ê</span>
                        <span class="star-label">5 min meditation</span>
                    </button>
                    <button class="star-btn" data-level="2" title="15 minutes self-care">
                        <span class="star-icon">‚≠ê</span>
                        <span class="star-label">15 min care</span>
                    </button>
                    <button class="star-btn" data-level="3" title="30 minutes wellness">
                        <span class="star-icon">‚≠ê</span>
                        <span class="star-label">30 min wellness</span>
                    </button>
                </div>
                <div class="task-streak">
                    <span class="task-streak-text">Streak: <span class="task-streak-count" id="caringStreak">0</span>
                        days</span>
                </div>
            </div>
        </main>

        <!-- Achievement Section -->
        <section class="achievements" id="achievementsSection" style="display: none;">
            <div class="achievement-popup" id="achievementPopup">
                <div class="achievement-content">
                    <div class="achievement-icon">üéâ</div>
                    <h3 class="achievement-title" id="achievementTitle"></h3>
                    <p class="achievement-message" id="achievementMessage"></p>
                </div>
            </div>
        </section>

        <!-- Weekly Summary (initially hidden) -->
        <section class="weekly-summary">
            <button class="summary-toggle" id="summaryToggle">
                <span>View Weekly Progress</span>
                <span class="toggle-icon">üìä</span>
            </button>
            <div class="summary-content" id="summaryContent" style="display: none;">
                <h3>This Week's Progress</h3>
                <div class="weekly-grid" id="weeklyGrid"></div>
            </div>
        </section>
    </div>

    <script src="script.js"></script>
</body>

</html>